### PR TITLE
MOB-2949 - Timeout driver allocation

### PR DESCRIPTION
**Pull request by jsm20**
## JIRA
[MOB-2949](https://karhoo.atlassian.net/browse/MOB-2949)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [ ] New feature 
- [x] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [x] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [x] Lint and unit tests pass locally (if appropriate)
- [x] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
Screenshot_1611075626

### CODE REVIEW

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/tripallocation/TripAllocationPresenter.kt [hunk 1]
```diff
@@ -62,11 +71,9 @@ class TripAllocationPresenter(view: TripAllocationMVP.View,
     }
 
     private fun tripAllocated(trip: TripInfo) {
-        tripDetailsObservable?.apply {
-            tripDetailsObserver?.let {
-                unsubscribe(it)
-            }
-        }
+        showAllocationAlert = false
+        showAllocationAlert = false
```
Comment (EdwardW2) [inline uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/tripallocation/TripAllocationPresenter.kt:75]: Is there a reason this is duplicated twice? "showAllocationAlert = false"

Comment (jsm20) [reply uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/tripallocation/TripAllocationPresenter.kt]: Deleted

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/tripallocation/TripAllocationView.kt [hunk 2]
```diff
@@ -66,6 +67,8 @@ class TripAllocationView @JvmOverloads constructor(
         cancelButton.isEnabled = true
         cancelButton.setListener { cancelTrip() }
         presenter?.waitForAllocation(trip)
+        handler.postDelayed({ presenter?.handleAllocationDelay() }, ALLOCATION_ALERT_DELAY)
```
Comment (mohadian) [inline uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/tripallocation/TripAllocationView.kt:70]: I wonder what happens when we are in the Guest mode and the user is navigated to the WebTracker but this timer still running, do you think we should schedule this thread only if it is not Guest mode?

Comment (jsm20) [reply uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/tripallocation/TripAllocationView.kt]: Done

**Review by mohadian — APPROVED** [review]

**Pull request merged**